packages/framework-integration-tests/integration/cli/cli.project.integration.ts

```diff
@@ -0,0 +1,205 @@
+import path = require('path')
+import util = require('util')
+import { expect } from 'chai'
+import { readFileContent } from '../helper/fileHelper'
+import * as fs from 'fs'
+import { ChildProcess, ExecException } from 'child_process'
+
+const exec = util.promisify(require('child_process').exec)
+
+const PROJECT_NAME_FIXTURE_PLACEHOLDER = 'project_name_fixture_placeholder'
+const TEST_TIMEOUT = 60000
+const DOWN_KEY = '\u001b[B'
+const DESCRIPTION = 'cart-demo'
+const VERSION = '1.0.0'
+const AUTHOR = 'The Agile Monkeys'
+const HOMEPAGE = 'https://www.booster.cloud/'
+const LICENSE = 'MIT'
+const REPO_URL = 'https://github.com/boostercloud/booster/'
+const PROVIDER = '@boostercloud/framework-provider-aws'
+const CART_DEMO_SHORT_FLAGS = 'cart-demo-short-flags'
+const CART_DEMO_LONG_FLAGS = 'cart-demo-long-flags'
+const CART_DEMO_COMMAND_PROMPT = 'cart-demo-command-prompt'
+const CART_DEMO_INVALID_PROVIDER = 'cart-demo-invalid-provider'
+const CART_DEMO_CUSTOM_PROVIDER = 'cart-demo-custom-provider'
+const CART_DEMO_FLAGS_AND_COMMAND_PROMPT = 'cart-demo-flags-and-command-prompt'
+
+export const CLI_PROJECT_INTEGRATION_TEST_FOLDERS: Array<string> = [
+  CART_DEMO_SHORT_FLAGS,
+  CART_DEMO_LONG_FLAGS,
+  CART_DEMO_COMMAND_PROMPT,
+  CART_DEMO_INVALID_PROVIDER,
+  CART_DEMO_CUSTOM_PROVIDER,
+  CART_DEMO_FLAGS_AND_COMMAND_PROMPT
+]
+
+describe('Project', () => {
+  const cliPath = path.join('..', 'cli', 'bin', 'run')
+  const expectedOutputRegex = new RegExp(
+    /(.+) boost (.+)?new(.+)? (.+)\n- Creating project root\n(.+) Creating project root\n- Generating config files\n(.+) Generating config files\n- Installing dependencies\n(.+) Installing dependencies\n(.+) Project generated!\n/
+  )
+
+  const sendToStdin = (childProcess: ChildProcess, promptAnswers: Array<string>, delay: number) => {
+    let currentRepetitions = 0;
+    const totalRepetitions = promptAnswers.length
+    let answers: Array<string> = promptAnswers.slice()
+
+    let intervalID = setInterval(() => {
+      childProcess.stdin?.write(answers[0])
+      answers = answers.slice(1)
+
+      if (++currentRepetitions === totalRepetitions) {
+        clearInterval(intervalID);
+      }
+    }, delay);
+  }
+
+  const execNewProject = (projectName: string, promptAnswers: Array<string> = [], flags: Array<string> = []) => {
+    return new Promise<string>((resolve): void => {
+      const childProcess = require('child_process').exec(`${cliPath} new:project ${projectName} ${flags.join(' ')}`,
```

Comment:
you could simplify this line if you add the code below to the top of the file
```
import util = require('util')
const exec = util.promisify(require('child_process').exec)
```

```suggestion
      const childProcess = exec(`${cliPath} new:project ${projectName} ${flags.join(' ')}`,
```

Reply:
The code already has that exec const at the top of the [file](https://github.com/boostercloud/booster/pull/184/files#diff-7aa788437077ac27a72747346874c349R8) and I used it in some tests, but this case is different. I don't want to use the `promisify` version here. Using that approach I was not able to write in `stdin` to provide answers to CLI's questions since I don't have access to the `stdin` until the Promise is resolved, but the Promise never resolves because it gets stuck waiting for a response that never comes. I got a timeout error in that scenario. That's why I create a new Promise manually here where I execute the bare `exec` and I am able to send stuff to `stdin`. Once the exec process finishes and its callback is executed, I can resolve the Promise with the stdout value. That was the tricky part I spent most of the time coding these tests.

Reply:
makes sense 👌